vortex-scalar/src/lib.rs

```diff
@@ -221,6 +222,13 @@ impl PartialOrd for Scalar {
     }
 }
 
+impl Hash for Scalar {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        discriminant(self.dtype()).hash(state);
+        self.value.0.hash(state);
```

Comment:
I don't think this is right... it requires the same internal scalar value representation in order to hash to the same thing.